src/v/utils/string_switch.h

```diff
@@ -182,7 +182,10 @@ class string_switch {
 
     constexpr operator R() { // NOLINT
         if (!result) {
-            throw std::runtime_error("Fell off the end of a string-switch");
+            throw std::runtime_error(
```

Comment:
Even better would be a message like `"foo" failed to match against ["bar", "baz", ...]` but accumulating the match candidates like that seems expensive, whereas the needle we already have it here and hopefully it's enough.